tests/unit/javascript/stylesheet/test_stylesheet.py

```diff
@@ -0,0 +1,124 @@
+from qutebrowser.browser.webengine import webenginesettings
+
+
+DEFAULT_BODY_BG = "rgba(0, 0, 0, 0)"
+GREEN_BODY_BG = "rgb(0, 255, 0)"
+CSS_BODY_GREEN = "body {background-color: rgb(0, 255, 0);}"
+CSS_BODY_RED = "body {background-color: rgb(255, 0, 0);}"
+
+
+class StylesheetTester:
+
+    """Helper class (for the stylesheet_tester fixture) for asserts.
+
+    Attributes:
+        js: The js_tester fixture.
+        config_stub: The config stub object.
+    """
+
+    def __init__(self, js_tester, config_stub):
+        self.js = js_tester
+        self.config_stub = config_stub
+
+    def init_stylesheet(self, css_file="green.css"):
+        """Initialize the stylesheet with a provided css file."""
+        self.config_stub.val.content.user_stylesheets = \
+            os.path.join(os.path.dirname(__file__), css_file)
+        p = QWebEngineProfile.defaultProfile()
+        webenginesettings._init_stylesheet(p)
+
+    def set_css(self, css):
+        """Set css to CSS via stylesheet.js."""
+        code = javascript.assemble('stylesheet', 'set_css', css)
+        self.js.run(code, None)
+
+    def check_set(self, value, element="background-color"):
+        """Check whether the css in ELEMENT is set to VALUE."""
+        self.js.run("window.getComputedStyle(document.body, null)"
+                    ".getPropertyValue('{}');".format(element), value)
+
+
+@pytest.fixture
+def stylesheet_tester(js_tester_webengine, config_stub):
+    """Helper fixture to test stylesheets."""
+    ss_tester = StylesheetTester(js_tester_webengine, config_stub)
+    ss_tester.js.webview.show()
+    return ss_tester
+
+
+@pytest.mark.parametrize('page', ['stylesheet/simple.html',
+                                  'stylesheet/simple_bg_set_red.html'])
+@pytest.mark.parametrize('set_js', [True, False])
+def test_set_delayed(stylesheet_tester, page, set_js):
+    """Test a delayed invocation of set_css."""
+    stylesheet_tester.init_stylesheet("none.css")
+    stylesheet_tester.js.load(page)
+    if set_js:
+        stylesheet_tester.js.run(
+            'document.body.style.backgroundColor = "red";', 'red')
+        pytest.xfail("overring values set with js does not work.")
+    stylesheet_tester.set_css("body {background-color: rgb(0, 255, 0);}")
+    stylesheet_tester.check_set("rgb(0, 255, 0)")
+
+
+@pytest.mark.parametrize('page', ['stylesheet/simple.html',
+                                  'stylesheet/simple_bg_set_red.html'])
+def test_set_clear_bg(stylesheet_tester, page):
+    """Test setting and clearing the stylesheet."""
+    stylesheet_tester.init_stylesheet()
+    stylesheet_tester.js.load('stylesheet/simple.html')
+    stylesheet_tester.check_set(GREEN_BODY_BG)
+    stylesheet_tester.set_css("")
+    stylesheet_tester.check_set(DEFAULT_BODY_BG)
+
+
+def test_no_set_xml(stylesheet_tester):
+    """Test stylesheet never modifies xml files."""
+    stylesheet_tester.init_stylesheet()
+    pytest.xfail("loading xml/svg files throws exceptions")
```

Comment:
xml/svg loading throws [a TypeError](https://github.com/qutebrowser/qutebrowser/pull/3212#issuecomment-343029708), ignoring that, the tests run fine.